File: Leetcode/Leetcode-DSA-course/Linked-List/20-valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        matching = {"(": ")", "{": "}", "[": "]"}
        for c in s:
            if c in matching:
                stack.append(c)
            else:
                if not stack:
                    return False
                previousOpening = stack.pop()
                if matching[previousOpening] != c:
                    return False
        return not stack

File: Leetcode/Leetcode-DSA-course/Linked-List/20-valid-parentheses/valid-parentheses.ts
function isValid(s: string): boolean {
    let stack = [];
    let matching = { '(': ')', '{': '}', '[': ']' };
    for (let c of s) {
        if (c in matching)
            stack.push(c);
        else {
            if (!stack.length)
                return false;
            let previousOpening = stack.pop()
            if (matching[previousOpening] != c)
                return false;
        }

    }
    return !stack.length
};